From: Sven Roederer <devel-sven@geroedel.de>
Date: Sat, 19 Sep 2020 22:50:46 +0200
Subject: build, imagebuilder: Do not require compilers

The buildroot and SDK both require the compilers (gcc, g++) to be
installed on the host system, however the ImageBuilder uses precompiled
binaries.

This patch changes the prerequirements checks to skip the checking for
the compilers if running as ImageBuilder. A similar change has been
made for libncurses-dev in 4a1a58a3e2d2.

Signed-off-by: Sven Roederer <devel-sven@geroedel.de>

diff --git a/include/prereq-build.mk b/include/prereq-build.mk
index 4637c6ca50bd6119e4adf269b57a87d70d522809..f67a01299e3a7018659a7baa89a0e9a9a1f08bfd 100644
--- a/include/prereq-build.mk
+++ b/include/prereq-build.mk
@@ -26,6 +26,7 @@ $(eval $(call TestHostCommand,proper-umask, \
 	Please build with umask 022 - other values produce broken packages, \
 	umask | grep -xE 0?0[012][012]))
 
+ifndef IB
 $(eval $(call SetupHostCommand,gcc, \
 	Please install the GNU C Compiler (gcc) 4.8 or later, \
 	$(CC) -dumpversion | grep -E '^(4\.[8-9]|[5-9]\.?|10\.?)', \
@@ -37,7 +38,9 @@ $(eval $(call TestHostCommand,working-gcc, \
 	it appears to be broken, \
 	echo 'int main(int argc, char **argv) { return 0; }' | \
 		gcc -x c -o $(TMP_DIR)/a.out -))
+endif
 
+ifndef IB
 $(eval $(call SetupHostCommand,g++, \
 	Please install the GNU C++ Compiler (g++) 4.8 or later, \
 	$(CXX) -dumpversion | grep -E '^(4\.[8-9]|[5-9]\.?|10\.?)', \
@@ -50,6 +53,7 @@ $(eval $(call TestHostCommand,working-g++, \
 	echo 'int main(int argc, char **argv) { return 0; }' | \
 		g++ -x c++ -o $(TMP_DIR)/a.out - -lstdc++ && \
 		$(TMP_DIR)/a.out))
+endif
 
 ifndef IB
 $(eval $(call TestHostCommand,ncurses, \
